src/ImageSharp/Common/Helpers/Vector128Utilities.cs

```diff
@@ -0,0 +1,174 @@
+    /// <returns>The <see cref="Vector128{Single}"/>.</returns>
+    public static Vector128<float> Shuffle(Vector128<float> vector, [ConstantExpected] byte control)
+    {
+        if (Sse.IsSupported)
+        {
+            return Sse.Shuffle(vector, vector, control);
+        }
+
+        ThrowUnreachableException();
+        return default;
+    }
+
+    /// <summary>
+    /// Creates a new vector by selecting values from an input vector using a set of indices.
+    /// </summary>
+    /// <param name="vector">
+    /// The input vector from which values are selected.</param>
+    /// <param name="indices">
+    /// The per-element indices used to select a value from <paramref name="vector" />.
+    /// </param>
+    /// <returns>
+    /// A new vector containing the values from <paramref name="vector" /> selected by the given <paramref name="indices" />.
+    /// </returns>
+    public static Vector128<byte> Shuffle(Vector128<byte> vector, Vector128<byte> indices)
+    {
+        if (Ssse3.IsSupported)
+        {
+            return Ssse3.Shuffle(vector, indices);
+        }
+
+        if (AdvSimd.Arm64.IsSupported)
+        {
+            return AdvSimd.Arm64.VectorTableLookup(vector, indices);
+        }
+
+        ThrowUnreachableException();
+        return default;
+    }
+
+    /// <summary>
+    /// Shifts a 128-bit value right by a specified number of bytes while shifting in zeros.
+    /// </summary>
+    /// <param name="value">The value to shift.</param>
+    /// <param name="numBytes">The number of bytes to shift by.</param>
+    /// <returns>The <see cref="Vector128{Byte}"/>.</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Vector128<byte> ShiftRightBytesInVector(Vector128<byte> value, [ConstantExpected(Max = (byte)15)] byte numBytes)
+    {
+        if (Sse2.IsSupported)
+        {
+            return Sse2.ShiftRightLogical128BitLane(value, numBytes);
+        }
+
+        if (AdvSimd.IsSupported)
+        {
+            return AdvSimd.ExtractVector128(value, Vector128<byte>.Zero, numBytes);
+        }
+
+        ThrowUnreachableException();
+        return default;
+    }
+
+    /// <summary>
+    /// Shifts a 128-bit value left by a specified number of bytes while shifting in zeros.
+    /// </summary>
+    /// <param name="value">The value to shift.</param>
+    /// <param name="numBytes">The number of bytes to shift by.</param>
+    /// <returns>The <see cref="Vector128{Byte}"/>.</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Vector128<byte> ShiftLeftBytesInVector(Vector128<byte> value, [ConstantExpected(Max = (byte)15)] byte numBytes)
+    {
+        if (Sse2.IsSupported)
+        {
+            return Sse2.ShiftLeftLogical128BitLane(value, numBytes);
+        }
+
+        if (AdvSimd.IsSupported)
+        {
+            return AdvSimd.ExtractVector128(Vector128<byte>.Zero, value, numBytes);
```

Comment:
```suggestion
            return AdvSimd.ExtractVector128(Vector128<byte>.Zero, value, 16 - numBytes);
```

This is the only thing that looks amiss to me.

Reply:
Yeah that's it.